blockchain/blockchain.go

```diff
@@ -275,6 +275,22 @@ func (td *stateTrieMigrationDB) ReadPreimage(hash common.Hash) []byte {
 	return td.ReadPreimageFromNew(hash)
 }
 
+func (bc *BlockChain) stateMigrationCommit(s *statedb.TrieSync, db database.DBManager) (int, time.Duration, error) {
+	start := time.Now()
+	stateTrieBatch := db.NewBatch(database.StateTrieDB)
+
+	written, err := s.Commit(stateTrieBatch)
+	if written == 0 || err != nil {
```

Comment:
Do we need to check `written == 0`?

Reply:
I get this code from present fast sync code. :)
I think this check to be able to prevent call `Write` with no item.